### PR TITLE
chore: sam sync & sam logs ux changes

### PR DESCRIPTION
**Pull request by mndeveci**
Following UX changes have been included;
- S3 bucket location and capabilities are not printed at the beginning of `sam sync` command since it is already printed later.
- During the build operation, UUIDs of the build definitions are replaced by the list of resource logical ids.
- Experimental warning color have changed to yellow.
- `sam logs` warning message regarding to missing SFN logging configuration is fixed
- `sam logs` error handling improvements
  - Proper error messaging is printed when no region is configured
  - Proper error messaging is printed when user token is expired
  - If `--stack-name` and `--cw-log-group` is not provided, error message shown that at least one of them is required.
  - Invalid `--stack-name` is handled with error messaging
  - Invalid CloudWatch log group or resource logical id (`--name`) values are added as warning message

#### Which issue(s) does this change fix?
N/A

#### Why is this change necessary?
To improve user experience with `sam sync` and `sam logs` commands.

#### What side effects does this change have?
N/A

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/commands/_utils/command_exception_handler.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    This function returns a wrapped function definition, which handles configured exceptions gracefully
+    """
+    def decorator_command_exception_handler(func):
+        @wraps(func)
+        def wrapper_command_exception_handler(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except Exception as ex:
+                exception_type = type(ex)
+                additional_mapping_with_defaults = (additional_mapping or {})
```
Comment (mndeveci) [inline samcli/commands/_utils/command_exception_handler.py:27]: nit: can call `get` at the end of this line here

File: samcli/commands/logs/validation_and_exception_handlers.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    """
+
+    @wraps(func)
+    def wrapped(*args, **kwargs):
+        ctx = click.get_current_context()
+        stack_name = ctx.params.get("stack_name")
+        cw_log_groups = ctx.params.get("cw_log_groups")
+
+        # either --stack-name or --cw-log-group flags should be provided
+        if not stack_name and not cw_log_groups:
```
Comment (mndeveci) [inline samcli/commands/logs/validation_and_exception_handlers.py:33]: `--stack-name` is required if `--name` is provided.

File: samcli/commands/sync/command.py [hunk 3]
```diff
@@ -265,6 +253,14 @@ def do_cli(
     set_experimental(ExperimentalFlag.Accelerate)
     update_experimental_context()
 
+    s3_bucket = manage_stack(profile=profile, region=region)
+
+    build_dir = DEFAULT_BUILD_DIR_WITH_AUTO_DEPENDENCY_LAYER if dependency_layer else DEFAULT_BUILD_DIR
+    LOG.debug("Using build directory as %s", build_dir)
+
+    build_dir = DEFAULT_BUILD_DIR_WITH_AUTO_DEPENDENCY_LAYER if dependency_layer else DEFAULT_BUILD_DIR
```
Comment (mndeveci) [inline samcli/commands/sync/command.py:261]: Remove duplicated line here

File: samcli/lib/build/build_graph.py [hunk 4]
```diff
@@ -527,6 +532,9 @@ def __init__(
         # this and move "layer" out of LayerBuildDefinition to take advantage of type check.
         self.layer: LayerVersion = None  # type: ignore
 
+    def get_resource_full_paths(self) -> str:
+        return self.layer.full_path if self.layer else self.uuid
```
Comment (mndeveci) [inline samcli/lib/build/build_graph.py:536]: - Return empty string here if layer is None
- Debug uuid if layer is None

**Review by mildaniel — APPROVED** [review]

**Pull request merged**